src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -8325,6 +8325,109 @@ public void PathMapParser()
             Assert.Equal(1, parsedArgs.Errors.Count());
             Assert.Equal((int)ErrorCode.ERR_InvalidPathMap, parsedArgs.Errors[0].Code);
         }
+
+        public class QuotedArgumentTests
+        {
+            private void VerifyQuotedValid<T>(string name, string value, T expected, Func<CSharpCommandLineArguments, T> getValue)
+            {
+                var args = DefaultParse(new[] { $"/{name}:{value}", "a.cs" }, @"c:\");
+                Assert.Equal(0, args.Errors.Length);
+                Assert.Equal(expected, getValue(args));
+
+                args = DefaultParse(new[] { $@"/{name}:""{value}""", "a.cs" }, @"c:\");
+                Assert.Equal(0, args.Errors.Length);
+                Assert.Equal(expected, getValue(args));
+            }
+
+            private void VerifyQuotedInvalid<T>(string name, string value, T expected, Func<CSharpCommandLineArguments, T> getValue)
+            {
+                var args = DefaultParse(new[] { $"/{name}:{value}", "a.cs" }, @"c:\");
+                Assert.Equal(0, args.Errors.Length);
+                Assert.Equal(expected, getValue(args));
+
+                args = DefaultParse(new[] { $@"/{name}:""{value}""", "a.cs" }, @"c:\");
+                Assert.True(args.Errors.Length > 0);
+            }
+
+            [WorkItem(12427, "https://github.com/dotnet/roslyn/issues/12427")]
+            [Fact]
+            public void DebugFlag()
+            {
+                var list = new List<Tuple<string, DebugInformationFormat>>()
+                {
+                    Tuple.Create("portable", DebugInformationFormat.PortablePdb),
+                    Tuple.Create("full", DebugInformationFormat.Pdb),
+                    Tuple.Create("pdbonly", DebugInformationFormat.Pdb),
+                    Tuple.Create("embedded", DebugInformationFormat.Embedded)
+                };
+
+                foreach (var tuple in list)
+                {
+                    VerifyQuotedValid("debug", tuple.Item1, tuple.Item2, x => x.EmitOptions.DebugInformationFormat);
+                }
+            }
```

Comment:
@jaredpar I don't have much experience either, but apart from the obvious ones (it's shorter, I think it's simpler and clearer), it also means it's easier to find the failure, if there's one.

If I try to force a failure here by adding a failing combination, the output without `[InlineData]` does not say why it failed:

```
Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests.CommandLineTests+QuotedArgumentTests.DebugFlag [FAIL]
  Assert.Equal() Failure
  Expected: 0
  Actual:   1
  Stack Trace:
    CommandLineTests.cs(8335,0): at Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests.CommandLineTests.QuotedArgumentTests.VerifyQuotedValid[T](String name, String value, T expected, Func`2 getValue)
    CommandLineTests.cs(8368,0): at Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests.CommandLineTests.QuotedArgumentTests.DebugFlag()
```

But with `[InlineData]`, you can clearly see that the failing parameters were `value: "pdb", expected: Pdb`

```
Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests.CommandLineTests+QuotedArgumentTests.DebugFlag(value: "pdb", expected: Pdb) [FAIL]
  Assert.Equal() Failure
  Expected: 0
  Actual:   1
  Stack Trace:
    CommandLineTests.cs(8335,0): at Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests.CommandLineTests.QuotedArgumentTests.VerifyQuotedValid[T](String name, String value, T expected, Func`2 getValue)
    CommandLineTests.cs(8362,0): at Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests.CommandLineTests.QuotedArgumentTests.DebugFlag(String value, DebugInformationFormat expected)
```